index.d.ts

```diff
@@ -0,0 +1,6301 @@
+         * @param name The name of the field to get.
+         */
+        data(name?: string): any;
+        /**
+         * Set a particular data field for the element.
+         * @param name The name of the field to set.
+         * @param value The value to set for the field.
+         */
+        data(name: string, value: any): this;
+        /**
+         * Update multiple data fields at once via an object.
+         * @param obj The object containing name- value pairs to update data fields.
+         */
+        data(obj: any): this;
+        /**
+         * Get a particular data field for the element.
+         * @param name The name of the field to get.
+         */
+        attr(name?: string): any;
+        /**
+         * Set a particular data field for the element.
+         * @param name The name of the field to set.
+         * @param value The value to set for the field.
+         */
+        attr(name: string, value: any): this;
+        /**
+         * Update multiple data fields at once via an object.
+         * @param obj The object containing name- value pairs to update data fields.
+         */
+        attr(obj: any): this;
+
+        /**
+         * Remove developer-defined data associated with the elements.
+         * http://js.cytoscape.org/#eles.removeData
+         * @param names A space-separated list of fields to delete.
+         */
+        removeData(names?: string): CollectionReturnValue;
+        removeAttr(names?: string): CollectionReturnValue;
+
+        /**
+         * Get an array of the plain JavaScript object
+         * representation of all elements in the collection.
+         */
+        jsons(): string[];
+    }
+    /**
+     * http://js.cytoscape.org/#collection/data
+     */
+    interface SingularData {
+        /**
+         * Get or set the scratchpad at a particular namespace,
+         * where temporary or non-JSON data can be stored.
+         * Get scratchpad if one or no parameter provided.
+         * App-level scratchpad data should use namespaces
+         * prefixed with underscore, like '_foo'.
+         * http://js.cytoscape.org/#ele.scratch
+         * @param namespace A namespace string.
+         * @param value The value to set at the specified namespace.
+         */
+        scratch(namespace?: string): Scratchpad;
+        scratch(namespace: string, value: any): this;
+
+        /**
+         * Remove scratchpad data.
+         * You should remove scratchpad data only at your own namespaces.
+         * http://js.cytoscape.org/#ele.removeScratch
+         * @param namespace A namespace string.
+         */
+        removeScratch(namespace: string): this;
+
+        /**
+         * A shortcut to get the ID of an element.
+         * http://js.cytoscape.org/#ele.id
+         */
+        id(): string;
+
+        /**
+         * Get the element's plain JavaScript object representation.
+         * http://js.cytoscape.org/#ele.json
+         */
+        json(): string;
```

Comment:
This does not return a string, it returns a JSON object.

Reply:
Right.  Someone must have confused a JSON string for a JSON object.  Good catch